src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/beans/identityprovider/IdentityProviderDetail.java

```diff
@@ -0,0 +1,38 @@
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.identityprovider;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.util.List;
+
+/**
+ * Created by chaoy01 on 2018-08-16.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes( {
+        @JsonSubTypes.Type(value=BindOnlyLdapIdentityProviderDetail.class, name="BIND_ONLY_LDAP"),
+        @JsonSubTypes.Type(value=LdapIdentityProviderDetail.class, name="LDAP")
+})
+public abstract class IdentityProviderDetail {
+
+    //type is used for JsonSubType annotations. Currently supports "BIND_ONLY_LDAP" and "LDAP"
+    private String type;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused "type" private field. ([squid:S1068](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1068))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=58&issues=AWVKO96biZQY_aXE1Ls0&open=AWVKO96biZQY_aXE1Ls0)

[ISSUE_KEY]: AWVKO96biZQY_aXE1Ls0